apps/site/components/Common/Button/index.tsx

```diff
@@ -17,17 +22,40 @@ const Button: FC<ButtonProps> = ({
   href = undefined,
   children,
   className,
+  onClick,
   ...props
-}) => (
-  <Link
-    role="button"
-    href={disabled ? undefined : href}
-    aria-disabled={disabled}
-    className={classNames(styles.button, styles[kind], className)}
-    {...props}
-  >
-    {children}
-  </Link>
-);
+}) => {
+  // Check if the component is being used as a button (no href provided)
+  const isButton = !href;
+
+  // Handle keydown events for keyboard accessibility
+  const handleKeyDown = (e: KeyboardEvent<HTMLAnchorElement>) => {
+    if (disabled) return;
+
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      if (onClick) {
+        onClick(e as unknown as MouseEvent<HTMLAnchorElement>);
+      }
+    }
+  };
+
+  return (
+    <Link
+      role={isButton ? 'button' : undefined} // Set role to 'button' only when used as a button
+      href={disabled ? undefined : href} // Disable the link when disabled
+      aria-disabled={disabled}
+      className={classNames(styles.button, styles[kind], className, {
+        [styles.disabled]: disabled, // Add disabled style when appropriate
+      })}
+      tabIndex={isButton ? (disabled ? -1 : 0) : undefined} // Remove from tab order if disabled for buttons
```

Comment:
should we be instead (or additionally) using <button disabled=true /> here - it supports skipping the tab order

https://codepen.io/bmuenzenmeyer/pen/KwPgNGe/9d52f7e51fbf7a037b499caed800da9e

Reply:
nevermind, I see it passing through from props

Reply:
the idea is when the button is disabled it should not be focusable